src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs

```diff
@@ -44,6 +49,38 @@ internal static class ClassifiedSpansAndHighlightSpanFactory
                 documentSpan.Document, documentSpan.SourceSpan, cancellationToken).ConfigureAwait(false);
         }
 
+        public static ClassifiedTextElement BuildClassifiedTextElementForClassifiedSpans(IEnumerable<ClassifiedSpan> classifiedSpans, ITextSnapshot snapshot, CancellationToken cancellationToken)
+        {
+            Contract.Assert(classifiedSpans != null && classifiedSpans.Any());
+
+            // Convert spans to textruns
+            var runs = new List<ClassifiedTextRun>();
+            var lastSpanEnd = -1;
+            foreach (var span in classifiedSpans)
+            {
+                // Add whitespace
+                if (lastSpanEnd > 0 && span.TextSpan.Start > lastSpanEnd)
+                {
+                    // find the gap between last span and current span, read the gap text from document
+                    var spanGap = new Span(lastSpanEnd, span.TextSpan.Start - lastSpanEnd);
+                    var spanText = snapshot.GetText(spanGap);
+                    runs.Add(new ClassifiedTextRun(ClassificationTypeNames.WhiteSpace, RemoveIndentsAfterNewLine(spanText)));
+                }
+
+                lastSpanEnd = span.TextSpan.End;
+                runs.Add(new ClassifiedTextRun(span.ClassificationType, snapshot.GetText(span.TextSpan.ToSpan())));
+            }
+
+            return new ClassifiedTextElement(runs);
+        }
+
+        private static string RemoveIndentsAfterNewLine(string text)
+        {
+            // use Contains to check new line instead of StartsWith, 
+            // so trailing whitespace(e.g. " \r\n    " from the previous line won't mess up the display
+            return text.Contains("\r\n") ? "\r\n" : text.Contains("\n") ? "\n" : text;
```

Comment:
this seems it will mess up all indentation and put everything to start from column  0? doesn't it actually need to find indentation of the first token on the first line, and remove that from all lines after?

can you link the existing code that handle quick info for this case? the existing code should do something like above to make sure unnecessary indentation to show up.